File: app/src/main/java/com/futsch1/medtimer/database/Tag.java
package com.futsch1.medtimer.database;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import com.google.gson.annotations.Expose;

import java.util.Objects;

@Entity
public class Tag {
    @PrimaryKey(autoGenerate = true)
    public int tagId;
    @Expose
    public String name;

    public Tag(String name) {
        this.name = name;
    }

    @Override
    public int hashCode() {
        return Objects.hash(tagId, name);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Tag tag = (Tag) o;
        return tagId == tag.tagId && Objects.equals(name, tag.name);
    }
}

File: app/src/main/java/com/futsch1/medtimer/database/Tag.kt
package com.futsch1.medtimer.database

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import com.google.gson.annotations.Expose
import java.util.Objects

@Entity
class Tag(@field:Expose var name: String) {
    @PrimaryKey(autoGenerate = true)
    var tagId: Int = 0

    @Ignore
    constructor() : this("")

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        return membersEqual(other as Tag)
    }

    override fun hashCode(): Int {
        return Objects.hash(tagId, name)
    }

    private fun membersEqual(tag: Tag): Boolean {
        return tagId == tag.tagId && name == tag.name
    }
}
